ixdiagnose/plugins/smart.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any
+
+from ixdiagnose.utils.middleware import MiddlewareClient, MiddlewareCommand
+from ixdiagnose.utils.run import run
+
+from .base import Plugin
+from .metrics import MiddlewareClientMetric, PythonMetric
+
+
+def smart_output(client: MiddlewareClient, context: Any) -> str:
+    include = '8,65,66,67,68,69,70,71,128,129,130,131,132,133,134,135,254,259'
+    cp = run(['lsblk', '-ndo', 'name', '-I', include], check=False)
+    if cp.returncode:
+        return cp.stderr
+
+    output = ''
+    for disk in filter(bool, cp.stdout.splitlines()):
+        command = ''
+        msg = '(NVME device detected)' if 'nvme' in disk else ''
+        msg = f'Block Device: /dev/{disk} {msg}'
+        command += f'echo "{"=" * (len(msg) + 5)}"\n'
+        command += f'echo "  {msg}"\n'
+        command += f'echo "{"=" * (len(msg) + 5)}"\n'
+        next_line = '\n' * 5
+        command += f'smartctl -a /dev/{disk}{next_line}'
+        cp = run(command, check=False, timeout=3)
```

Comment:
* `timeout=3` will raise an exception if the timeout is exceeded
* you pass a shell command, but you don't specify `shell=True`
* why not do all these echoes in Python?
* smartctl has JSON output (`-j`)

Reply:
@themylogin `timeout` is managed appropriately?
The program won't crash and we would move on to the next disk in question similar to how shell script does.

Changes were made to automatically see if the command requires `shell=True` based on last feedback..

And yes i'll make changes to do the echoes in python

Reply:
Sorry, I thought it was raw `subprocess.run`. The only concerns left are echoes and/or parseable output